### DISCUSSION TITLE
ASIO4ALL settings unreachable

### DISCUSSION THREAD

**Discussion by Matzix** [jamulussoftware/jamulus#1056] [UI Related]
I was experimenting a little bit using ASIO4ALL driver and by mistake, I used an output device with a sample rate other than 48kHz. In the consequence, Jamulus told me, that it has to be restarted and now ASIO4ALL can not be selected any more:
![grafik](https://user-images.githubusercontent.com/15308822/83136339-92436000-a0e7-11ea-95db-627a17776a38.png)
In the consequence, also the ASIO4ALL settings can not be opened any more from within Jamulus in order to fix these settings.

Is it possible to change the current behaviour, such that invalid soundcard settings prevents connecting to a server (in conjunction with an error message), but still be able to choose an invalid ASIO entry in setup when not connected?

**Comment by WolfganP** [comment]
As a workaround you may try to open the settings via another app to correct the offending settings, ie https://github.com/jprjr/asioconfig

**Comment by Matzix** [comment]
ASIO4ALL also provides a standalone GUI (choose "Off-Line Settings" in the installer). Nevertheless, in my test Jamulus didn't accept ASIO4ALL any more until I had rebootet my machine. It's not a critical issue, but something, which might be fixed some day.

**Comment by gilgongo** [comment]
See also https://sourceforge.net/p/llcon/discussion/developerforum/thread/bc8c5a5f7b/#7337

**Comment by ann0see** [comment]
Is this issue fixed? I think it's possible to at least change the device now. 
Ok. I can still reproduce this error. I think, there was a change made by @corrados which allowed you to use the ASIO settings?

**Comment by npostavs** [comment]
This looks the same as issue as #117 which is kinda fixed now by giving an option to open the ASIO control panel in case of an error (I think it's still a confusing UX though).

> ASIO4ALL also provides a standalone GUI (choose "Off-Line Settings" in the installer).

I think this changes only some kind of "global" settings, but ASIO4ALL maintains per-application settings so it has no effect on the settings that Jamulus uses. (I guess this is why the installer calls the off-line thing "useless").

**Comment by ann0see** [comment]
Probably all the ASIO functionality etc. has to be re-thought. I think it would be great if you could tweak Jamulus settings even if the ASIO driver is not configured correctly. This would also completely solve this issue.

**Comment by gilgongo** [comment]
Moving to a discussion until we can figure out what to do on this.
